protocol/triple/server.go

```
@@ -247,8 +247,24 @@ func (s *Server) handleServiceWithInfo(interfaceName string, invoker protocol.In
 					attachments := generateAttachments(req.Header())
 					// inject attachments
 					ctx = context.WithValue(ctx, constant.AttachmentKey, attachments)
+					oldAttachments := map[string]interface{}{}
+					if original, ok := ctx.Value(constant.AttachmentKey).(map[string]interface{}); ok {
+						for k, v := range original {
+							oldAttachments[k] = v
+						}
+					}
 					invo := invocation.NewRPCInvocation(m.Name, args, attachments)
 					res := invoker.Invoke(ctx, invo)
+					newAttachments, _ := ctx.Value(constant.AttachmentKey).(map[string]interface{})
```

do not ignore this error, may cause panic.